factory/vm/sovereignVmContainerShardCreator_test.go

```diff
@@ -13,17 +15,50 @@ import (
 func TestNewSovereignVmContainerShardCreatorFactory(t *testing.T) {
 	t.Parallel()
 
-	bhhc := &factory.BlockChainHookHandlerFactoryMock{}
-	sovereignVmContainerShardFactory, err := vm.NewSovereignVmContainerShardFactory(bhhc)
-	require.Nil(t, err)
-	require.False(t, sovereignVmContainerShardFactory.IsInterfaceNil())
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+		bhhc := &factory.BlockChainHookHandlerFactoryMock{}
+		runTypeComponents := componentsMock.GetRunTypeComponents()
+		sovereignVmContainerShardFactory, err := vm.NewSovereignVmContainerShardFactory(bhhc, runTypeComponents.VmContainerMetaFactoryCreator(), runTypeComponents.VmContainerShardFactoryCreator())
+		require.Nil(t, err)
+		require.False(t, sovereignVmContainerShardFactory.IsInterfaceNil())
+	})
+
+	t.Run("nil blockChainHookHandlerCreator", func(t *testing.T) {
+		t.Parallel()
+
+		sovereignVmContainerShardFactory, err := vm.NewSovereignVmContainerShardFactory(nil, nil, nil)
```

Comment:
Please use nil only for first argument(blockchainHool) and mocks/components for the rest